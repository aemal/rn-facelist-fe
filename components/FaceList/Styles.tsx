import { StyleSheet } from 'react-native';

export const FaceListStyles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: '#fff',
    alignItems: 'center',
    justifyContent: 'center',
  },
  item: {
    display: 'flex',
    flexDirection: 'row',
    alignItems: 'center',
    padding: 20,
    width: '100vw',
  },
  name: {
    marginLeft: 10
  },
  avatar: {
    width: 50,
    height: 50,
    borderWidth: 2,
    borderColor: '#3f51b5',
    borderRadius: 50
  }
});
