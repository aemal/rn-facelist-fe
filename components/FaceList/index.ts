export {default as FaceList} from './FaceList';
