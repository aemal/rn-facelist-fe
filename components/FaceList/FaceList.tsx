import { StatusBar } from 'expo-status-bar';
import React, { useState, useEffect } from 'react';
import { connect } from 'react-redux';
import { ThunkDispatch } from "redux-thunk";
import { AppState } from "../../redux/store";
import { AppActions } from "../../redux/types";
import { Text, View, FlatList, Image } from 'react-native';
import { FaceListStyles } from './Styles';
import { fetchUsers } from '../../redux/actions/usersAction';
import { IUser, IUsers } from '../../redux/types/userTypes';

interface IProps {
  users: IUser[];
  fetchUsers: any;
}

const FaceList: React.FC<IProps> = ({users, fetchUsers}) => {

  useEffect(() => {
    fetchUsers()
  });

  return (
    <View style={FaceListStyles.container}>
      <FlatList
        data={users}
        renderItem={({item}) => {
          return (
            <View style={FaceListStyles.item}>
              <Image style={FaceListStyles.avatar}
              source={{ uri: item.avatar }} />
              <Text style={FaceListStyles.name}>{item.name}</Text>
            </View>
          )
        }}
      />
    </View>
  );
}

const mapStateToProps = (state: AppState) => {
  return {
    users: state.users.usersList
  };
}

const mapDispatchToProps = (dispatch: ThunkDispatch<AppState, undefined, AppActions>,) => {
  return {
    fetchUsers: () => dispatch(fetchUsers())
  };
}

export default connect(mapStateToProps, mapDispatchToProps)(FaceList);
