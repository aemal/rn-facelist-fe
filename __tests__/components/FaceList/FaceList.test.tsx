import thunk from "redux-thunk";
import { FaceList } from "../../../components/FaceList";
import renderer from 'react-test-renderer';
import React from 'react';
import fetchMock from 'fetch-mock'
import configureMockStore from "redux-mock-store";
const middleware = [thunk];
const mockStore = configureMockStore(middleware);

//mock data
const initialState = {
  users: {
    usersList: [
      { "id": "1", "avatar": "https://i.pravatar.cc/150?img=1", "name": "Morgan James" },
      { "id": "2", "avatar": "https://i.pravatar.cc/150?img=2", "name": "Esme Coleman" },
      { "id": "3", "avatar": "https://i.pravatar.cc/150?img=3", "name": "Gary Watson" },
    ]
  },
};

//mock api for component
const mockSuccessfulResponse = (
  status = 200,
  method = "GET",
  returnBody = initialState.users.usersList
) => {
  fetchMock.get("http://localhost:4000/facelist", () => ({
    method,
    response: {
      status: 200,
      body: JSON.stringify(returnBody)
    }
  }));
};

//store initialization
let store = mockStore(initialState);

//test case for FaceList component
test('FaceList Render Test',async () => {
  await mockSuccessfulResponse()
  const tree = renderer.create(<FaceList store={store} />).toJSON();
  expect(tree).toMatchSnapshot();
});