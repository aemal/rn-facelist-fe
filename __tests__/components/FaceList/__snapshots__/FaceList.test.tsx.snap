// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FaceList Render Test 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#fff",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <RCTScrollView
    data={
      Array [
        Object {
          "avatar": "https://i.pravatar.cc/150?img=1",
          "id": "1",
          "name": "Morgan James",
        },
        Object {
          "avatar": "https://i.pravatar.cc/150?img=2",
          "id": "2",
          "name": "Esme Coleman",
        },
        Object {
          "avatar": "https://i.pravatar.cc/150?img=3",
          "id": "3",
          "name": "Gary Watson",
        },
      ]
    }
    disableVirtualization={false}
    getItem={[Function]}
    getItemCount={[Function]}
    horizontal={false}
    initialNumToRender={10}
    keyExtractor={[Function]}
    maxToRenderPerBatch={10}
    onContentSizeChange={[Function]}
    onEndReachedThreshold={2}
    onLayout={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    removeClippedSubviews={false}
    renderItem={[Function]}
    scrollEventThrottle={50}
    stickyHeaderIndices={Array []}
    updateCellsBatchingPeriod={50}
    viewabilityConfigCallbackPairs={Array []}
    windowSize={21}
  >
    <View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "display": "flex",
              "flexDirection": "row",
              "padding": 20,
              "width": "100vw",
            }
          }
        >
          <Image
            source={
              Object {
                "uri": "https://i.pravatar.cc/150?img=1",
              }
            }
            style={
              Object {
                "borderColor": "#3f51b5",
                "borderRadius": 50,
                "borderWidth": 2,
                "height": 50,
                "width": 50,
              }
            }
          />
          <Text
            style={
              Object {
                "marginLeft": 10,
              }
            }
          >
            Morgan James
          </Text>
        </View>
      </View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "display": "flex",
              "flexDirection": "row",
              "padding": 20,
              "width": "100vw",
            }
          }
        >
          <Image
            source={
              Object {
                "uri": "https://i.pravatar.cc/150?img=2",
              }
            }
            style={
              Object {
                "borderColor": "#3f51b5",
                "borderRadius": 50,
                "borderWidth": 2,
                "height": 50,
                "width": 50,
              }
            }
          />
          <Text
            style={
              Object {
                "marginLeft": 10,
              }
            }
          >
            Esme Coleman
          </Text>
        </View>
      </View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "display": "flex",
              "flexDirection": "row",
              "padding": 20,
              "width": "100vw",
            }
          }
        >
          <Image
            source={
              Object {
                "uri": "https://i.pravatar.cc/150?img=3",
              }
            }
            style={
              Object {
                "borderColor": "#3f51b5",
                "borderRadius": 50,
                "borderWidth": 2,
                "height": 50,
                "width": 50,
              }
            }
          />
          <Text
            style={
              Object {
                "marginLeft": 10,
              }
            }
          >
            Gary Watson
          </Text>
        </View>
      </View>
    </View>
  </RCTScrollView>
</View>
`;
